Add an overview/intro for the Library API

crankyoldgit opened the pull request:
* Force docs/doxygen_index.md to be used as main page for the Doxygen documentation
* Initial version of doxygen_index.md

crankyoldgit committed:
Add an overview/intro for the Library API
* Force docs/doxygen_index.md to be used as main page for the Doxygen documentation
* Initial version of doxygen_index.md

NiKiZe reviewed (approved):
Should there be links to the doxys for kTolerance and similar instead?

crankyoldgit committed:
Fix typos and grammar.

crankyoldgit commented:
> Should there be links to the doxys for kTolerance and similar instead?

Most of it doxygen autodetects and adds links. for `kTolerance` etc. it doesn't work. I don't think it works for enum items and constants, but it does for structures and classes. e.g. IRrecv & state_t etc

At this stage, I'm not too worried. I just a want something other than a blank page at the root of the doxygen tree. ;-)

We can perfect it later. I also suspect as you've come to this library more recently than I, you'll have a better perspective of where to point people to, to get the most out of it (the library's APIs and documentation). I have to vast and detailed a viewpoint to see it with "new person" eyes.

crankyoldgit committed:
Fix doxygen warning.
`/home/travis/build/crankyoldgit/IRremoteESP8266/docs/doxygen_index.md:338: warning: explicit link request to 'define' could not be resolved`

Travis's doxygen flagged this, my local one doesn't. Grrr.

crankyoldgit commented:
> > Should there be links to the doxys for kTolerance and similar instead?

Also, this is what the URL looks like for `kTolerance` in doxygen: `IRremoteESP8266/docs/doxygen/html/IRrecv_8h.html#a7884008b3a738dfc7bd8658655e10272` I don't have faith that's going to work long-term. If we want it to work, there's probably some doxygen voodoo that we need to do, which I haven't sussed out yet.

NiKiZe commented:
Sounds good, didn't know ;)
